server/index.js

```diff
@@ -29,7 +29,7 @@ const blockedPages = {
 }
 
 export default class Server {
-  constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false, conf = null } = {}) {
+  constructor ({ dir = '.', dev = false, generateEtags = true, staticMarkup = false, quiet = false, conf = null } = {}) {
```

Comment:
Lets move the `generateEtags` to `server/config.js` https://github.com/zeit/next.js/blob/canary/server/config.js#L5

You can then use it like `this.nextConfig.generateEtags`

And define `generateEtags` in next.config.js.